scripts/scil_search_keywords.py

```diff
@@ -55,164 +76,156 @@ def main():
+            if filename not in matches:
+                matches.append(filename)
+                scores[filename] = score_details
+            else:
+                for key, value in score_details.items():
+                    if key != 'total_score':
+                        scores[filename][key] = scores[filename].get(
+                            key, 0) + value
+                scores[filename]['total_score'] += score_details['total_score']
+
+    for script in sorted(script_dir.glob(search_pattern.format(selected_object))):
+        filename = script.stem
+        if filename == '__init__' or filename == 'scil_search_keywords':
             continue
 
-        # Skip this script
-        if filename == pathlib.Path(__file__).name:
+        # Search through the docstring
+        search_text = _get_docstring_from_script_path(str(script))
+        score_details = _calculate_score(
+            stemmed_keywords, stemmed_phrases, search_text, filename=filename)
+        update_matches_and_scores(filename, score_details)
+
+        # Search in help files
+        help_file = hidden_dir / f"{filename}.py.help"
+        if help_file.exists():
+            with open(help_file, 'r') as f:
+                search_text = f.read()
+            score_details = _calculate_score(
+                stemmed_keywords, stemmed_phrases,
+                search_text, filename=filename)
+            update_matches_and_scores(filename, score_details)
+
+    # Search in keywords file
+    with open(VOCAB_FILE_PATH, 'r') as f:
+        vocab_data = json.load(f)
+
+    for script in vocab_data['scripts']:
+        script_name = script['name']
+        if selected_object and not script_name.startswith(f'scil_{selected_object}_'):
             continue
-
-        error_msg = ""
-        if args.search_parser:
-            # Run the script's argparser, and catch the output in case there
-            # is an error, such as ModuleNotFoundException.
-            sub = subprocess.run(['{}'.format(script.absolute()), '--help'],
-                                 capture_output=True)
-            search_text = sub.stdout.decode("utf-8")
-            if sub.stderr:
-                # Fall back on the docstring in case of error
-                error_msg = "There was an error executing script parser, " \
-                            "searching through docstring instead...\n\n"
+        script_keywords = script['keywords']
+        search_text = ' '.join(script_keywords)
+        score_details = _calculate_score(
+            stemmed_keywords, stemmed_phrases, search_text, script_name)
+        update_matches_and_scores(script_name, score_details)
+
+    # Search in synonyms file if not args.no_synonyms is not specified
+    if not args.no_synonyms:
+        for keyword in keywords + phrases:
+            synonyms = _get_synonyms(keyword, vocab_data['synonyms'])
+            for script in sorted(script_dir.glob(search_pattern.format(selected_object))):
```

Comment:
Not sure why at line 169 you do a for loop on `vocab_data['scripts']`, now you do a for loop on all scripts using glob, I believe the two for loops could be <merged> but the `if not args.no_synonyms:` be in that single loop. You would save a few lines